zenoh/src/net/protocol/network.rs

```diff
@@ -1159,3 +1195,104 @@ pub(crate) fn shared_nodes(net1: &Network, net2: &Network) -> Vec<ZenohIdProto>
         })
         .collect()
 }
+
+#[derive(Debug, Clone, PartialEq, serde::Serialize)]
+pub(crate) struct NetworkPathNode {
+    pub(crate) zid: ZenohIdProto,
+    pub(crate) locators: Option<Vec<Locator>>,
```

Comment:
I believe this is not the place to put those informations. I'd rather have the paths be simple lists of zids and let an eventual tooling get the locators of each node if needed.